docs/guide/hosting.md

```diff
@@ -0,0 +1,46 @@
+# Hosting Apostrophe in production
+
+## Server requirements
+
+We recommend hosting Apostrophe in self-hosted production environments with the following minimum specifications:
+
+| Software | Minimum Version |
+| ------------- | ------------- |
+| Ubuntu | 16+ |
+| [NGINX](https://www.nginx.com/) (or another proxy server) | Latest |
+| [Node.js](https://nodejs.org/en/) | 12.x+ |
+| [MongoDB](https://docs.mongodb.com/manual/administration/install-community/)  | 3.6+ |
+| [ImageMagick](https://imagemagick.org/script/download.php#macosx) (optional) | Any |
+
+
+| Hardware specification | Minimum recommendation |
+| ------------- | ------------- |
+| RAM | **2GB** (to support the application and database) |
+| Disk space | **20GB** (see note below) |
+
+1GB of RAM may be sufficient if using a remote database service (e.g., MongoDB Atlas). Sites running in a multicore configuration will require additional 2G or more RAM and, of course, CPU cores.
+
+::: note
+MongoDB requires a minimum of 5GB free disk space at all times. Small websites may not need much more additional space than that. We have found that 20GB is sufficient for most Apostrophe applications, but more space may be necessary to handle an especially large number of file uploads or especially large uploaded files.
+:::
+
+## Deployment basics
+
+Deployment processes will vary depending on the hosting environment, technical requirements, and team preferences. There are a few steps that all deployment processes should include.
```

Comment:
NODE_ENV should be set to `production` at all times in production and production-like environments (such as staging servers). This ensures that performance enhancements are enabled.